### PR TITLE
Adding null checks before FileTag method (fix fatal)

### PR DESCRIPTION
**Pull request by nupplaphil**
Fixing https://github.com/friendica/friendica/issues/6916#issuecomment-497054109
FollowUp #7199

### CODE REVIEW

File: src/Model/FileTag.php [hunk 1]
```diff
@@ -98,15 +102,19 @@ public static function arrayToFile(array $array, string $type = 'file')
...
 	 *
-	 * @param string $file File tags
-	 * @param string $type Optional file type.
+	 * @param string|null $file File tags
+	 * @param string      $type Optional file type.
 	 *
 	 * @return array        List of tag names.
 	 */
-	public static function fileToArray(string $file, string $type = 'file')
+	public static function fileToArray($file, string $type = 'file')
```
Comment (MrPetovan) [inline src/Model/FileTag.php]: I'd rather if you left the strict definition of this method and instead changed `include/text.php:189` with:
```php
foreach (FileTag::fileToArray($item['file'] ? '', 'category') as $savedFolderName) {
```
And `include/text.php:205`
```php
foreach (FileTag::fileToArray($item['file'] ?? '') as $savedFolderName) {
```

File: src/Model/FileTag.php [hunk 2]
```diff
@@ -68,13 +68,17 @@ public static function fileQuery($table, $s, $type = 'file')
...
 	 *
-	 * @param array  $array A list of tags.
-	 * @param string $type  Optional file type.
+	 * @param array|null $array A list of tags.
+	 * @param string     $type  Optional file type.
 	 *
 	 * @return string       A list of file tags.
 	 */
-	public static function arrayToFile(array $array, string $type = 'file')
+	public static function arrayToFile($array, string $type = 'file')
```
Comment (MrPetovan) [inline src/Model/FileTag.php]: I'd rather if you left the strict definition of this method.

**Comment by AlfredSK** [comment]
@MrPetovan Are you change requests resolved?

File: mod/item.php [hunk 3]
```diff
@@ -337,7 +337,7 @@ function item_post(App $a) {
 	$categories = FileTag::listToFile(trim(defaults($_REQUEST, 'category', '')), 'category');
 	$categories_new = $categories;
 
-	if (!empty($filedas)) {
+	if (!empty($filedas) && is_array($filedas)) {
```
Comment (MrPetovan) [inline mod/item.php]: `$filedas` is only returned by `FileTag::fileToArray` that only returns array, did you get a specific type error? I'd rather you initialized `$filedas = []` before line 330.

**Comment by MrPetovan** [comment]
Apparently force-pushing doesn't trigger a notification on my side.

**Pull request merged**